### PR TITLE
Homework 43 completed

### PR DESCRIPTION
**Pull request by Borisko130**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
-target/*
\ No newline at end of file
+target/*
+logs/*
```
Comment (MaliukDaria) [inline .gitignore:4]: add an empty line at the end of the file

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -20,24 +22,25 @@
             = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
     private static OrderService orderService
             = (OrderService) injector.getInstance(OrderService.class);
+    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/Main.java:25]: why CinemaHallDaoImpl.class?

File: src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -9,16 +9,19 @@
...
 public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Logger logger = Logger.getLogger(AuthenticationServiceImpl.class);
     @Inject
     private UserService userService;
     @Inject
     private ShoppingCartService shoppingCartService;
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
+        logger.info("User with email " + email + " logs in");
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java:24]: what if there is no such user in DB?

File: src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -28,6 +31,7 @@ public User login(String email, String password) throws AuthenticationException
 
     @Override
     public User register(String email, String password) {
+        logger.info("Registering new User");
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java:34]: maybe you should add some information about registering user? (e-mail for example)

File: .gitignore (hunk 1 above)
Comment (KseniiaMakarova) [reply .gitignore:4]: I agree

File: src/main/resources/log4j.properties [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Root logger options
+log4j.rootLogger=DEBUG, CONSOLE, FILE
+
+# CONSOLE appender customisation
+log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
+log4j.appender.CONSOLE.threshold=DEBUG
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties:6]: let's add log4j.appender.stdout.Target

File: src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java (hunk 3 above)
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/security/impl/AuthenticationServiceImpl.java:24]:
```suggestion
        logger.info("User with email " + email + " tries to log in");
```

File: src/main/java/com/dev/cinema/Main.java (hunk 2 above)
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:25]:
```suggestion
    private static final Logger logger = Logger.getLogger(Main.class);
```

File: src/main/java/com/dev/cinema/Main.java [hunk 6]
```diff
@@ -20,24 +22,25 @@
...
 
     public static void main(String[] args) throws AuthenticationException {
         authenticationService.register("email@mail.net", "1234");
         User user = userService.findByEmail("email@mail.net").get();
         MovieSession movieSession = new MovieSession();
         shoppingCartService.addSession(movieSession, user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:31]: where are movies, moviesessions, users and other entities saved to DB before we can proceed?

File: src/main/java/com/dev/cinema/Main.java [hunk 7]
```diff
@@ -20,24 +22,25 @@
...
 
-        System.out.println("FIRST ORDER");
+        logger.info("FIRST ORDER");
         ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
-        System.out.println(orderService.completeOrder(shoppingCart.getTickets(),
+        logger.debug(orderService.completeOrder(shoppingCart.getTickets(),
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:35]: let's use `info` level here everywhere

File: src/main/java/com/dev/cinema/Main.java [hunk 8]
```diff
@@ -20,24 +22,25 @@
             = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
     private static OrderService orderService
             = (OrderService) injector.getInstance(OrderService.class);
+    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
 
     public static void main(String[] args) throws AuthenticationException {
         authenticationService.register("email@mail.net", "1234");
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:28]: let's wrap call to `login()` method in try-catch and add log.warn with the informative message in `catch` block

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/resources/log4j.properties [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+log4j.appender.CONSOLE.layout=org.apache.log4j.PatternLayout
+log4j.appender.CONSOLE.layout.ConversionPattern=[%-5p] : %c:%L : %m%n
+
+# Direct log messages to stdout
+log4j.appender.STDOUT=org.apache.log4j.ConsoleAppender
+log4j.appender.STDOUT.Target=System.out
+log4j.appender.STDOUT.layout=org.apache.log4j.PatternLayout
```
Comment (Borisko130) [inline src/main/resources/log4j.properties:13]: Not sure whether log4j.appender.CONSOLE should be replaced with log4j.appender.STDOUT so left both

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**